pkg/rpc/server/listen.go

```diff
@@ -49,10 +53,14 @@ func ListenAndServe(c config.Config, fsCache cache.FSCache) error {
 
 	go func() {
 		worker := initializeDBWorker(c.CacheDir, true)
+		if err := initializeMetricGauge(c.GaugeMetric, c.CacheDir); err != nil {
+			c.GaugeMetric = nil // just in case some if some error/issue occurs (probably with GaugeMetric). So, making c.GaugeMetric nil and hence, unusable for any of the functions which require it
```

Comment:
This is not needed. In go by default the value will be `nil`. You've added a check before using it in functions where it's used and that's sufficient.